docs/DeveloperGuide.md

```diff
@@ -1,34 +1,182 @@
+`StorageFile` achieves this by using two helper classes, `TrackingListEncoder` and `TrackingListDecoder`.
+
+`TrackingListEncoder` takes the currently stored `TrackingList` and converts it to a more storage friendly String.
+This String is then written to a file on the disk, known on the diagram as `StoredTrackingList`.
+As the name suggests, `TrackingListDecoder` achieves the reverse by taking the file on drive, `StoredTrackingList`,
+and interprets the file to be loaded into the memory as a `TrackingList`.
+
+`StoredTrackingList` is the `.txt` file stored in the same directory as the program.
+The location is determined by the `ConfigFile` class. 
+
+`LogFile` takes the `TrackingList` every time the storage is updated
+and packages it into `json` formatted string.
+`LogFile` communicates with the disk by calling the methods
+`loadLogFile` and `saveLogFile` in the `StorageFile` class.
+
+
+The program settings for the user is manged by the `ConfigFile` class. 
+Another file, `settingsFile` is read by `ConfigFile` when the program is started, 
+to load the settings into the program memory. 
+When a change is made by the user, `ConfigFile` will update the `settingsFile` through a save function.
+
+### Common classes
+
+Classes used by multiple components are in the `seedu.Duke.commons` package. 
+
+## Implementation
+
+## Documentation
+
+## Testing
+
+## Dev Ops
+
+## Appendix
+
+### Appendix A: Product scope
+[WIP]
+
+#### Target user profile
+* User that requires more control over crowd level management in a building.
+* Can type fast
+* Prefers a simple desktop application over the more complicated systems
+* Is reasonably comfortable with CLI apps
+* Requires real time monitoring and tracking capability of people in the building
+* Wants a cheaper alternative to crowd tracking systems
 
 {Describe the target user profile}
 
-### Value proposition
+#### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+A fast and versatile software solution that enables mall management to crowd control 
+in the most efficient way possible.
 
-## User Stories
+### Appendix B: User Stories 
+[WIP]
 
-|Version| As a ... | I want to ... | So that I can ...|
+|Priority| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|*****|mall staff|be able to review the personal particulars of the customer|be aware of who he is|
+|*****|new user|be able to use the program without much training| use the program as soon as possible and easily implement it|
+|****|mall staff|key in personal particulars easily into the application| check in visitors quickly without holding up a line of customers|
+|****|mall staff|be able to easily key in a unique personal identifier| check in and find a visitor|
```

Comment:
I like how priorities are added to the user stories to show the importance. But it seems like more user stories can be added